### PR TITLE
MM: Add mm iface to progress queue only if there is a remote connection.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1175/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@alinask can you pls review?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1177/ for details (Mellanox internal link).

File: src/uct/sm/mm/mm_iface.h [hunk 1]
```diff
@@ -93,7 +97,10 @@ struct uct_mm_fifo_element {
 
 struct uct_mm_fifo_ctl {
     volatile uint64_t  head;       /* where to write next */
-    char padding[UCS_SYS_CACHE_LINE_SIZE - sizeof(uint64_t)];
+    socklen_t          signal_addrlen;
+    struct sockaddr_un signal_sockaddr;
+    char               null;
+    UCS_CACHELINE_PADDING(uint64_t, socklen_t, struct sockaddr_un, char);
```
Comment (shamisp) [inline src/uct/sm/mm/mm_iface.h]: @yosefe , as I read it, it does not have the same effect as the original code

Comment (yosefe) [inline src/uct/sm/mm/mm_iface.h]: how come? it's a generalization of adding a padding element to a structure..

File: src/uct/sm/mm/mm_ep.c [hunk 2]
```diff
@@ -66,6 +96,11 @@ static UCS_CLASS_CLEANUP_FUNC(uct_mm_ep_t)
     uct_mm_remote_seg_t *remote_seg;
     struct sglib_hashed_uct_mm_remote_seg_t_iterator iter;
 
+    uct_mm_ep_send_connect(self, 0);
```
Comment (shamisp) [inline src/uct/sm/mm/mm_ep.c]: Define 1/0 as connect/disconnect

Comment (yosefe) [inline src/uct/sm/mm/mm_ep.c]: as enum?

Comment (shamisp) [inline src/uct/sm/mm/mm_ep.c]: enum even better.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1789/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1180/ for details (Mellanox internal link).

File: src/uct/sm/mm/mm_iface.c [hunk 3]
```diff
@@ -300,7 +301,16 @@ ucs_status_t uct_mm_allocate_fifo_mem(uct_mm_iface_t *iface,
...
 
-    uct_mm_set_fifo_ptrs(iface->shared_mem, &iface->recv_fifo_ctl, &iface->recv_fifo_elements);
+    uct_mm_set_fifo_ptrs(iface->shared_mem, &ctl, &iface->recv_fifo_elements);
+
+    /* Make sure head and tail are cahe-aligned, and not on same cacheline, to
+     * avoid false-sharing.
+     */
+    ucs_assert((((uintptr_t)&ctl->head) % UCS_SYS_CACHE_LINE_SIZE) == 0);
+    ucs_assert((((uintptr_t)&ctl->tail) % UCS_SYS_CACHE_LINE_SIZE) == 0);
+    ucs_assert(((uintptr_t)&ctl->tail - (uintptr_t)&ctl->head) >= UCS_SYS_CACHE_LINE_SIZE);
```
Comment (shamisp) [inline src/uct/sm/mm/mm_iface.c]: ucs_always_assert ?

Comment (yosefe) [inline src/uct/sm/mm/mm_iface.c]: Ok

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1791/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1182/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
depends on #738

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1793/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1187/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1188/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1794/ for details.

**Comment by yosefe** [comment]
depends on #739

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1205/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1811/ for details.

**Comment by MattBBaker** [comment]
ORNL error:
```
[----------] 3 tests from test_time, where TypeParam = 
[ RUN      ] test_time.get_time
/var/lib/jenkins/jobs/ornl_pod9/workspace/contrib/../test/gtest/ucs/test_time.cc:50: Failure
Expected: (nsec) < (30.0), actual: 30.105 vs 30
ucs_get_time() performance is too bad
[  FAILED  ] test_time.get_time, where TypeParam =  and GetParam() =  (1000 ms)
```
Seems silly. We should bump the threshold a bit.

**Comment by yosefe** [comment]
@MattBBaker how easily can this be reproduced (using GTEST_FILTER to run this test only)?
Can you please test with #744?

**Comment by yosefe** [comment]
@alinask IMB passed on one node 16 ppn (after merge with master)

**Comment by alinask** [comment]
:+1:

**Pull request merged**